cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -761,25 +761,25 @@ public void checkCustomCaSecret(CertificateAuthority ca, Secret certSecret, Secr
                             if (featureGates.useStrimziPodSetsEnabled())   {
                                 return strimziPodSetOperator.getAsync(namespace, KafkaCluster.kafkaClusterName(name))
                                         .compose(podSet -> {
-                                            if (podSet != null)    {
-                                                return Future.succeededFuture(podSet.getSpec().getPods().size());
+                                            if (podSet != null) {
+                                                return Future.succeededFuture(KafkaCluster.generatePodList(reconciliation.name(), podSet.getSpec().getPods().size()));
                                             } else {
-                                                return Future.succeededFuture(0);
+                                                return Future.succeededFuture(KafkaCluster.generatePodList(reconciliation.name(), 0));
```

Comment:
Should this just be `List.of()`?

Reply:
List.of() will return me a list of `object` type(List<E>) while the `generatePodList` podList method is returning me a List of string type(List<String) and since this if else is later on followed by a compose so the `future` returned should be of one type (List<E> or List<String>)

Reply:
Fair enough ... but I guess you can still do something like `new ArrayList<String>(0)`?